src/participant/participant.ts

```diff
@@ -1605,82 +1619,61 @@ export default class ParticipantController {
   }
 
   async exportCodeToPlayground(): Promise<boolean> {
-    const activeTextEditor = vscode.window.activeTextEditor;
-    if (!activeTextEditor) {
-      await vscode.window.showErrorMessage('Active editor not found.');
-      return false;
-    }
+    const selectedText = getSelectedText();
+    const codeToExport = selectedText || getAllText();
 
-    const sortedSelections = Array.from(activeTextEditor.selections).sort(
-      (a, b) => a.start.compareTo(b.start)
-    );
-    const selectedText = sortedSelections
-      .map((selection) => activeTextEditor.document.getText(selection))
-      .join('\n');
-    const code =
-      selectedText || activeTextEditor.document.getText().trim() || '';
     try {
-      const progressResult = await vscode.window.withProgress(
+      const content = await vscode.window.withProgress(
         {
           location: vscode.ProgressLocation.Notification,
           title: 'Exporting code to a playground...',
           cancellable: true,
         },
         async (progress, token): Promise<string | null> => {
           const modelInput = await Prompts.exportToPlayground.buildMessages({
-            request: { prompt: code },
+            request: { prompt: codeToExport },
           });
 
           const result = await Promise.race([
-            this.getChatResponseContent({
+            this.streamChatResponseWithExportToLanguage({
               modelInput,
               token,
             }),
-            new Promise<undefined>((resolve) =>
+            new Promise<null>((resolve) =>
               token.onCancellationRequested(() => {
-                resolve(undefined);
+                void vscode.window.showInformationMessage(
+                  'The export to a playground operation was canceled.'
```

Comment:
This adds another click for users, since they are the one who clicked the cancel action I'm thinking we can just silently resolve here. How does removing this sound?

Reply:
We match the playground cancelation case. Should we remove it there as well?

Reply:
I'd say yes, let's ask Gaurab. I'll create a slack thread. What are your thoughts on it? I'm also thinking back to when we showed notifications for connecting which people found annoying and created prs to turn off lol.